src/Components/Assets/AssetsList.tsx

```diff
@@ -116,7 +116,14 @@ const AssetsList = () => {
         const { data } = await request(routes.listAssets, {
           query: { qr_code_id: assetId },
         });
-        return data?.results[0].id;
+        if (data?.results.length === 0) {
+          setIsLoading(false);
+          Notification.Error({
+            msg: "You are not authorized to view this asset",
+          });
+        } else {
+          return data?.results[0].id;
```

Comment:
Also, if it is just to get a single asset, consider using get API instead of list

Reply:
The get API doesn't have qr search facility so that's the reason the list is used

Reply:
https://careapi.ohc.network/swagger/#/asset_qr/api_v1_public_asset_qr_retrieve

Reply:
@rithviknishad when we are using listAsset API if we are in the wrong login it gives 0 results but where as getQR API returns it even if the user is not authorized 

So instead of no assets it will show invalid QR

Reply:
<img width="510" alt="image" src="https://github.com/coronasafe/care_fe/assets/98876115/f5eddcaa-79e5-4b15-8ab5-8642c2730868">


It shows error here saying not found

Reply:
![image](https://github.com/coronasafe/care_fe/assets/98876115/920a2e85-4125-4f0a-9f0c-2f2a73b1c8fe)
here it shows error saying not found 404

Reply:
then let's continue using the list api itself since that worked so far...
Instead pass query parameter: `limit: 1` to fetch only 1 item max.